tests/unit/test_ena_queries.py

```diff
@@ -0,0 +1,99 @@
+import responses
+
+from assembly_uploader.ena_queries import EnaQuery
+
+
+def test_ena_query_study(study_data):
+    responses.add(
+        responses.POST,
+        "https://www.ebi.ac.uk/ena/portal/api/search",
+        json=[
+            {
+                "study_accession": "PRJEB41657",
+                "study_title": "HoloFood Salmon Trial A+B Gut Metagenome",
+                "first_public": "2022-08-02",
+            }
+        ],
+    )
+
+    responses.add(
+        responses.GET,
+        "https://www.ebi.ac.uk/ena/submit/report/studies/ERP125469",
+        json=[
+            {
+                "report": {
+                    "id": "ERP125469",
+                    "alias": "ena-STUDY-UNIVERSITY OF COPENHAGEN-02-12-2020-08:09:52:762-111",
+                    "firstCreated": "2020-12-02T08:09:55",
+                    "firstPublic": "2022-08-02T17:21:21",
+                    "releaseStatus": "PUBLIC",
+                    "submissionAccountId": "Webin-51990",
+                    "secondaryId": "PRJEB41657",
+                    "title": "HoloFood Salmon Trial A+B Gut Metagenome",
+                    "holdDate": "null",
+                },
+                "links": [],
+            }
+        ],
+    )
+
+    ena_study_public = EnaQuery(
+        accession="ERP125469",
+        private=False,
+    )
+
+    ena_study_private = EnaQuery(
+        accession="ERP125469",
+        private=True,
+    )
+
+    assert (
+        ena_study_public.build_query() and ena_study_private.build_query() == study_data
+    )
+
+
+def test_ena_query_run(run_public, run_private):
```

Comment:
I think we should just add one more test for the retry logic. So, adding a response that returns empty data or a bad status code or whatever, and then check that the URL is called 3 times:

```python
responses.add(
        responses.GET,
        "https://www.ebi.ac.uk/ena/submit/report/studies/ERP125469",
        json=None
)
with pytest.raises(NoDataException):
    EnaQuery(accession="ERP125469", private=False)
responses.assert_call_count("https://www.ebi.ac.uk/ena/submit/report/studies/ERP125469", 3)
```
or something along those lines